File: June/19-06-2024/Find maximum volume of a cuboid.js
//{ Driver Code Starts

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split("\n").map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {

        let input_p = readLine().split(' ').map(x => parseFloat(x));
        let p = input_p[0];
        let s = input_p[1];

        let obj = new Solution();
        let res = obj.maxVolume(p, s);

        console.log(parseFloat(res).toFixed(2));
    }
}

// } Driver Code Ends



class Solution {
    /**
    * @param float p
    * @param float s

    * @returns float
    */
    maxVolume(perimeter, area) {
       //Your Code Start
             // Calculate the length of one side of the base
        let len = (perimeter - Math.sqrt(perimeter * perimeter - 24 * area)) / 12.0;
        
        // Calculate the volume of the cuboid
        let vol = len * len * ((perimeter / 4.0) - 2 * len);
        
        // Round the volume to two decimal places
        return Math.round(vol * 100) / 100;
      //Your Code End
    }
}

File: June/19-06-2024/Find maximum volume of a cuboid.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    double maxVolume(double perimeter, double area) {
      //Your Code Start
       // Calculate the length of one side of the base
        double len = (perimeter - sqrt((perimeter * perimeter) - 24 * area)) / 12.0;
        
        // Calculate the volume of the cuboid
        double vol = len * len * ((perimeter / 4.0) - 2 * len);
        
        // Round the volume to two decimal places
        return round(vol * 100.0) / 100.0;
      //Your Code End
    }
};


//{ Driver Code Starts.
int main() {
    long long int t;
    cin >> t;
    while (t--) {
        double perimeter, area;
        cin >> perimeter >> area;
        // if (area == 5 and parameter == 15) {
        //     cout << "0.46" << endl;
        //     return 0;
        // }
        Solution ob;
        double ans = ob.maxVolume(perimeter, area);
        cout.precision(2);
        cout << fixed << ans << "\n";
    }
}
// } Driver Code Ends
